### PR TITLE
Implement more useful empty state for overview page

### PR DESCRIPTION
**Pull request by TheRealJon**
https://jira.coreos.com/browse/CONSOLE-1146

With at least one project available:
image

With no projects available:
image

@openshift/team-ux-review

### CODE REVIEW

File: frontend/public/style/_overrides.scss [hunk 1]
```diff
@@ -203,3 +203,12 @@ tags-input .autocomplete .suggestion-item em {
     border-bottom: 1px solid $table-border-color;
   }
 }
+
+.blank-slate-pf {
```
Comment (spadgett) [inline frontend/public/style/_overrides.scss]: If we are adding this override, we should be able to remove at least some of the styles here:

https://github.com/spadgett/console/blob/0b573b72939e2500af46e22e2a11ef921c8a871c/frontend/public/components/catalog/_catalog.scss#L172-L180

File: frontend/public/components/overview/index.tsx [hunk 2]
```diff
@@ -275,6 +276,32 @@ const sortBuilds = (builds: K8sResourceKind[]): K8sResourceKind[] => {
...
+      <Link to="/catalog" className="btn btn-lg btn-primary">
+        Browse Catalog
+      </Link>
+    </EmptyState.Action>
+    <EmptyState.Action secondary>
+      <Link className="btn btn-default" to={`/deploy-image?preselected-ns=${activeNamespace}`} title="Deploy Image">
+        Deploy Image
+      </Link>
+      <Link className="btn btn-default" to={formatNamespacedRouteForResource('import', activeNamespace)} title="Perform an action">
+        Import YAML/JSON
```
Comment (spadgett) [inline frontend/public/components/overview/index.tsx]: I think we should just say Import YAML. JSON is a YAML subest, so it's technically YAML. Import YAML would also be consistent with the add dropdown in the project bar.

File: frontend/public/components/overview/index.tsx [hunk 3]
```diff
@@ -275,6 +276,32 @@ const sortBuilds = (builds: K8sResourceKind[]): K8sResourceKind[] => {
...
+    <EmptyState.Action>
+      <Link to="/catalog" className="btn btn-lg btn-primary">
+        Browse Catalog
+      </Link>
+    </EmptyState.Action>
+    <EmptyState.Action secondary>
+      <Link className="btn btn-default" to={`/deploy-image?preselected-ns=${activeNamespace}`} title="Deploy Image">
+        Deploy Image
+      </Link>
+      <Link className="btn btn-default" to={formatNamespacedRouteForResource('import', activeNamespace)} title="Perform an action">
```
Comment (spadgett) [inline frontend/public/components/overview/index.tsx]: I feel like there should be a little more margin between the buttons.

File: frontend/public/components/overview/index.tsx [hunk 4]
```diff
@@ -275,6 +276,32 @@ const sortBuilds = (builds: K8sResourceKind[]): K8sResourceKind[] => {
...
 
+const overviewEmptyStateToProps = ({UI}) => ({activeNamespace: UI.get('activeNamespace')});
+
+const OverviewEmptyState = connect(overviewEmptyStateToProps)(({activeNamespace}) => (
+  <EmptyState>
+    <EmptyState.Title>
+      Get started with your project.
+    </EmptyState.Title>
+    <EmptyState.Info>
+      Add content to your project from the catalog of web frameworks, databases, and other components. You may also deploy an existing image or create or replace resources using YAML and JSON definitions.
```
Comment (spadgett) [inline frontend/public/components/overview/index.tsx]: Replace isn't supported.

```suggestion
      Add content to your project from the catalog of web frameworks, databases, and other components. You may also deploy an existing image or create resources using YAML definitions.
```

**Comment by spadgett** [comment]
Does this message show up when you have content, but it's filtered? We might want to show something else in that case.

File: frontend/public/style/_overrides.scss (hunk 1 above)
Comment (TheRealJon) [reply frontend/public/style/_overrides.scss]: I actually don't see that class being used anywhere in catalog, so those can probably be removed altogether unless I'm missing something.

File: frontend/public/components/overview/index.tsx (hunk 3 above)
Comment (TheRealJon) [reply frontend/public/components/overview/index.tsx]: This is actually defined in our overrides:

https://github.com/openshift/console/blob/d081e2fa76f4f166ae78dcb93195fe29ffebaa89/frontend/public/style/_overrides.scss#L148-L151

Do we want to adjust this, or create a more specific rule just for empty state secondary actions?

Comment (spadgett) [reply frontend/public/components/overview/index.tsx]: OK, let's leave it as-is for now.

**Comment by TheRealJon** [comment]
@spadgett This empty state message will now only show when no overview resources exist. The message `No Resources found.`  is shown when they are all filtered out.

File: frontend/public/components/utils/tile-view-page.jsx [hunk 5]
```diff
@@ -615,8 +615,8 @@ export class TileViewPage extends React.Component {
   renderEmptyState() {
     const { emptyStateTitle, emptyStateInfo } = this.props;
     return (
-      <EmptyState className="co-catalog-page__no-filter-results">
-        <EmptyState.Title className="co-catalog-page__no-filter-results-title" aria-level="2">
+      <EmptyState>
```
Comment (TheRealJon) [inline frontend/public/components/utils/tile-view-page.jsx]: oops, need to add the class names back here

**Comment by spadgett** [comment]
jenkins rebuild

**Pull request merged**